src/common/android/deviceResolver.ts

```diff
@@ -0,0 +1,37 @@
+import {ChildProcess} from "../node/childProcess";
+import * as Q from "q";
+
+export interface IDevice {
+    id: string;
+    status: string;
+}
+
+export class DeviceResolver {
+
+    /**
+     * Gets the list of Android connected devices and emulators.
+     */
+    public getConnectedDevices(): Q.Promise<IDevice[]> {
```

Comment:
I think we should use a library for this.
These two seems reasonable candidates:
* https://www.npmjs.com/package/adbkit-logcat
* https://www.npmjs.com/package/adbjs
And I think we can use the licenses (I used https://github.com/davglass/license-checker and https://www.npmjs.com/package/nlf to make a quick check)

* There is also this one: https://www.npmjs.com/package/appium-adb but it has a looooooooot of dependencies, and some have some weird stuff inside, so I'm guessing that's probably not an option.